.codegen/service.go.tmpl

```diff
@@ -78,7 +78,7 @@ var {{.CamelName}}Overrides []func(
 func new{{.PascalName}}() *cobra.Command {
 	cmd := &cobra.Command{}
 
-	{{- $useJsonForAllFields := or .IsJsonOnly (and .Request (or (not .Request.IsAllRequiredFieldsPrimitive) .Request.HasRequiredNonBodyField))  -}}
+	{{- $useJsonForAllFields := or .IsJsonOnly (and .Request (or (not .Request.IsAllRequiredFieldsPrimitive) (not .Request.HasRequiredNonBodyField)))  -}}
```

Comment:
This logic is getting complex. Could we abstract this into the OpenAPI generator with a clearer heuristic?

Reply:
https://github.com/databricks/databricks-sdk-go/pull/661